src/firefly/js/ui/tap/TableSearchMethods.jsx

```diff
@@ -546,14 +698,87 @@ function tapSearchMethodReducer(columnsModel) {
                                 rFields.polygoncoords = clone(inFields.polygoncoords, {value: v});
                             }
                         }
+                    } else if (fieldKey === TimeFrom || fieldKey === TimeTo) {
```

Comment:
switch statements with a lot lof logic like this are hard to read. I think the whole sections under `FieldGroupCntlr.VALUE_CHANGE:` needs to be broken out into 2 or 3 functions. It is just too difficult to determine what is going on.

Reply:
thanks. updated.